sky/packages/sky/lib/rendering/object.dart

```diff
@@ -407,12 +407,14 @@ abstract class RenderObject extends AbstractNode implements HitTestTarget {
     sky.tracing.begin('RenderObject.flushLayout');
     _debugDoingLayout = true;
     try {
-      List<RenderObject> dirtyNodes = _nodesNeedingLayout;
-      _nodesNeedingLayout = new List<RenderObject>();
-      dirtyNodes..sort((a, b) => a.depth - b.depth)..forEach((node) {
-        if (node._needsLayout && node.attached)
-          node.layoutWithoutResize();
-      });
+      while(_nodesNeedingLayout.isNotEmpty) {
```

Comment:
Please add a `// TODO(ianh): assert that we're not allowing previously dirty nodes to redirty themeselves` here.